src/dune_util/report_error.ml

```diff
@@ -135,7 +135,7 @@ let format_memo_stack pps =
                    (Pp.seq (Pp.verbatim "-> ")
                       (Pp.seq (Pp.text "required by ") pp))))))
 
-let report { Exn_with_backtrace.exn; backtrace } =
+let gen_report exn backtrace =
```

Comment:
Looking at the call sites, perhaps `?backtrace exn` would be slightly more idiomatic.

Reply:
Did you see my other suggestion? I wonder if making `Exn_with_backtrace.t` an exception itself would be good. There's a few other situations where we provide a junk backtrace just because the type required is `Exn_with_backtrace.t` rather than `Exn.t`

Reply:
Sorry, I didn't fully understand that other suggestion. I think I'll need to see the code first. Happy to leave the type as is, if you think we're going to refactor this all anyway.

Reply:
It's quite similar to your suggestion. Basically, if the backtrace can be optional in many situations, we might as well change `Exn_with_backtrace.t` to:

```ocaml
type t =
  { exn : Exn.t
   ; backtrace : Printexc.backtrace option
  }
```

But then this type becomes equivalent to extending the exception type with a non optional back trace:

```ocaml
exception of Exn_with_backtrace of Exn_with_backtrace.t (* here the backtrace isn't optional *)
```

With this change, we can just `exn` everywhere and not require callers to provide fake backtraces.

Reply:
I see, makes sense.